In [136]:
import pandas as pd
import numpy as np
from scipy.stats import pearsonr
import matplotlib.pyplot as plt

import sys
sys.path.append('..')

from config import load_data

In [137]:
data_dir = "./data/"
target_variable = 'Q_Kalltveit'

In [138]:
ld = load_data(data_dir = data_dir, target_variable = target_variable)
data = ld.get_raw_data()

In [139]:
# Rename columns to be more user-friendly
data = data.rename(columns={
    'Vindhastighet Nilsebu': 'Wind Speed Nilsebu',
    'Lufttemp. Nilsebu': 'Air Temperature Nilsebu',
    'Vindretning Nilsebu': 'Wind Direction Nilsebu',
    'RelHum Nilsebu': 'Relative Humidity Nilsebu',
    'Vannstand Lyngsåna': 'Water Level Lyngsåna',
    'Vanntemp. Hiafossen': 'Water Temperature Hiafossen',
    'Vannstand Hiafossen': 'Water Level Hiafossen',
    'Lufttemp Fister': 'Air Temperature Fister',
    'Nedbør Fister': 'Precipitation Fister',
    'Q_Lyngsvatn_overlop': 'Flow Lyngsvatn Overflow',
    'Q_tapping': 'Flow Tapping',
    'Vannstand Kalltveit': 'Water Level Kalltveit',
    'Q_Kalltveit': 'Flow Kalltveit',
    'Vanntemp. Kalltveit kum': 'Water Temperature Kalltveit Kum',
    'Nedbør Nilsebu': 'Precipitation Nilsebu',
    'Vanntemp. Hiavatn': 'Water Temperature Hiavatn',
    'Vannstand Hiavatn': 'Water Level Hiavatn',
    'Vanntemp. Musdalsvatn': 'Water Temperature Musdalsvatn',
    'Vannstand Musdalsvatn': 'Water Level Musdalsvatn',
    'Vanntemp. Musdalsvatn nedstrøms': 'Water Temperature Musdalsvatn Downstream',
    'Vannstand Musdalsvatn nedstrøms': 'Water Level Musdalsvatn Downstream',
    'Vanntemp. Viglesdalsvatn': 'Water Temperature Viglesdalsvatn',
    'Vannstand Viglesdalsvatn': 'Water Level Viglesdalsvatn'
})

In [140]:
def get_correlation_matrix(data):
    # Calculate the partial correlation matrix
    corr_matrix = data.corr()
    partial_corr_matrix = pd.DataFrame(np.zeros_like(corr_matrix), columns=corr_matrix.columns, index=corr_matrix.index)
    for i in range(len(partial_corr_matrix.columns)):
        for j in range(i+1, len(partial_corr_matrix.index)):
            control_variables = [x for x in corr_matrix.index if x not in [partial_corr_matrix.index[j], partial_corr_matrix.columns[i]]]
            partial_corr = pearsonr(data[partial_corr_matrix.index[j]] - data[control_variables].mean(axis=1), data[partial_corr_matrix.columns[i]] - data[control_variables].mean(axis=1))[0]
            partial_corr_matrix.iloc[j, i] = partial_corr
            partial_corr_matrix.iloc[i, j] = partial_corr

    # Plot the partial correlation matrix as a heatmap
    plt.figure(figsize=(10,8))
    sns.heatmap(partial_corr_matrix, cmap='coolwarm', annot=True, fmt='.2f')
    plt.title('Partial Correlation Matrix of Hydrological Inflow Variables')
    plt.show()

In [141]:
data.describe().T

count        mean         std  \
Wind Speed Nilsebu                        62955.0  -11.359088  373.072147   
Air Temperature Nilsebu                   63102.0    3.636294    6.919231   
Wind Direction Nilsebu                    63079.0  213.033360  113.673486   
Relative Humidity Nilsebu                 63079.0   64.444010   43.178686   
Water Level Lyngsåna                      50380.0    0.735474    0.182148   
Water Temperature Hiafossen               50192.0    6.242513    5.352072   
Water Level Hiafossen                     50192.0    1.078482    0.158591   
Air Temperature Fister                    63288.0    8.811811    5.909750   
Precipitation Fister                      63288.0    0.215365    0.746016   
Flow Lyngsvatn Overflow                   63286.0    0.035255    0.742100   
Flow Tapping                              63286.0    0.941535    1.440778   
Water Level Kalltveit                     63274.0  -26.232780  517.624893   
Flow Kalltveit                            63288.0    6.509245    7.565526   
Water Temperature Kalltveit Kum           63269.0  -25.726374  563.037133   
Precipitation Nilsebu                     63288.0    0.326407    1.237064   
Water Temperature Hiavatn                 45837.0    6.713851    5.499990   
Water Level Hiavatn                       45836.0    1.944471    0.219938   
Water Temperature Musdalsvatn             45978.0    5.820647    5.436186   
Water Level Musdalsvatn                   45978.0    1.509856    0.289519   
Water Temperature Musdalsvatn Downstream  45955.0    5.851968    5.203346   
Water Level Musdalsvatn Downstream        45955.0    1.229056    0.149924   
Water Temperature Viglesdalsvatn          45858.0    6.728865    5.431330   
Water Level Viglesdalsvatn                45858.0    1.787619    0.186923   
Q_HBV                                     63288.0    5.353519    6.811531   
PRECIP_HBV                                63288.0    0.281887    0.821245   
TEMP_HBV                                  63288.0    4.070589    6.919953   
SNOW_MELT_HBV                             29595.0    0.088353    0.159276   
SNOW_SWE_HBV                              29595.0  187.212226  248.752510   
Evap_HBV                                  29595.0    0.037146    0.052701   
SOIL_WAT_HBV                              29595.0   71.159486   18.556167   
GR_WAT_HBV                                29595.0   15.950311   11.308697   
Q_Kalltveit_uten_tapping                  63288.0    5.589386    7.764969   
Q_HBV_mean                                31241.0    5.177136    6.595386   
Q_Lyngsaana                               63288.0    2.259291    3.344124   
Vanntemp. Lyngsåna                        63283.0    6.402902    5.343528   
Vanntemp. Kalltveit elv                   50316.0    7.309757    5.477978   

                                                  min         25%         50%  \
Wind Speed Nilsebu                       -7999.000000    5.200000    6.100000   
Air Temperature Nilsebu                    -20.700000   -0.800000    3.100000   
Wind Direction Nilsebu                     -99.900000  133.200000  215.600000   
Relative Humidity Nilsebu                  -99.900000   51.500000   82.600000   
Water Level Lyngsåna                         0.331000    0.606000    0.714000   
Water Temperature Hiafossen                 -0.200000    1.200000    4.900000   
Water Level Hiafossen                        0.849000    0.972000    1.029000   
Air Temperature Fister                     -11.800000    4.600000    8.400000   
Precipitation Fister                         0.000000    0.000000    0.000000   
Flow Lyngsvatn Overflow                      0.000000    0.000000    0.000000   
Flow Tapping                                 0.000000    0.000000    0.512519   
Water Level Kalltveit                    -9999.000000    0.526380    0.579130   
Flow Kalltveit                               0.000000    2.683190    4.097010   
Water Temperature Kalltveit Kum          -9999.000000    4.5

In [142]:
data.count().sort_values()

GR_WAT_HBV                                  29595
SOIL_WAT_HBV                                29595
Evap_HBV                                    29595
SNOW_SWE_HBV                                29595
SNOW_MELT_HBV                               29595
Q_HBV_mean                                  31241
Water Level Hiavatn                         45836
Water Temperature Hiavatn                   45837
Water Level Viglesdalsvatn                  45858
Water Temperature Viglesdalsvatn            45858
Water Level Musdalsvatn Downstream          45955
Water Temperature Musdalsvatn Downstream    45955
Water Temperature Musdalsvatn               45978
Water Level Musdalsvatn                     45978
Water Temperature Hiafossen                 50192
Water Level Hiafossen                       50192
Vanntemp. Kalltveit elv                     50316
Water Level Lyngsåna                        50380
Wind Speed Nilsebu                          62955
Wind Direction Nilsebu                      63079


The following is the variables with the number of values:

| Variable                                 | Count   |
| ---------------------------------------- | ------- |
| Q_Lyngsaana                              | 63288   |
| Q_Kalltveit_uten_tapping                 | 63288   |
| TEMP_HBV                                 | 63288   |
| Precipitation Nilsebu                    | 63288   |
| PRECIP_HBV                               | 63288   |
| Q_HBV                                    | 63288   |
| Precipitation Fister                     | 63288   |
| Flow Kalltveit                           | 63288   |
| Air Temperature Fister                   | 63288   |
| Datetime                                 | 63288   |

The following variables have some missing values:

| Variable                                 | Count   |
| ---------------------------------------- | ------- |
| Flow Tapping                             | 63286   |
| Flow Lyngsvatn Overflow                  | 63286   |

The following variables have varying values:

| Variable                                 | Count   |
| ---------------------------------------- | ------- |
| Wind Speed Nilsebu                       | 62955   |
| Wind Direction Nilsebu                   | 63079   |
| Relative Humidity Nilsebu                | 63079   |
| Air Temperature Nilsebu                  | 63102   |
| Water Temperature Kalltveit Kum          | 63269   |
| Water Level Kalltveit                    | 63274   |
| Vanntemp. Lyngsåna                       | 63283   |

The following variables have varying water varables:

| Variable                                 | Count   |
| ---------------------------------------- | ------- |
| Water Level Hiavatn                      | 45836   |
| Water Temperature Hiavatn                | 45837   |
| Water Level Viglesdalsvatn               | 45858   |
| Water Temperature Viglesdalsvatn         | 45858   |
| Water Level Musdalsvatn Downstream       | 45955   |
| Water Temperature Musdalsvatn Downstream | 45955   |
| Water Temperature Musdalsvatn            | 45978   |
| Water Level Musdalsvatn                  | 45978   |
| Water Temperature Hiafossen              | 50192   |
| Water Level Hiafossen                    | 50192   |
| Vanntemp. Kalltveit elv                  | 50316   |
| Water Level Lyngsåna                     | 50380   |

HBV simulation data:

| Variable                                 | Count   |
| ---------------------------------------- | ------- |
| GR_WAT_HBV                               | 29595   |
| SOIL_WAT_HBV                             | 29595   |
| Evap_HBV                                 | 29595   |
| SNOW_SWE_HBV                             | 29595   |
| SNOW_MELT_HBV                            | 29595   |
| Q_HBV_mean                               | 31241   |


In [143]:
var_with_max_count = ["Q_Lyngsaana", "Q_Kalltveit_uten_tapping", "TEMP_HBV", "Precipitation Nilsebu", "PRECIP_HBV", "Q_HBV", "Precipitation Fister", "Flow Kalltveit", "Air Temperature Fister", "Datetime"]                                  

In [144]:
data[var_with_max_count].describe()

Q_Lyngsaana  Q_Kalltveit_uten_tapping      TEMP_HBV  \
count  63288.000000              63288.000000  63288.000000   
mean       2.259291                  5.589386      4.070589   
std        3.344124                  7.764969      6.919953   
min        0.000000                  0.000000    -20.196868   
25%        0.469784                  1.299561     -0.383786   
50%        1.058914                  3.496592      3.577349   
75%        2.529453                  6.673359      9.052494   
max       55.227400                110.256270     27.716214   

       Precipitation Nilsebu    PRECIP_HBV         Q_HBV  \
count           63288.000000  63288.000000  63288.000000   
mean                0.326407      0.281887      5.353519   
std                 1.237064      0.821245      6.811531   
min                 0.000000      0.000000      0.005526   
25%                 0.000000      0.000000      1.829147   
50%                 0.000000      0.000000      3.436150   
75%                 0.098519      0.137533      6.331031   
max                30.000000     26.323696     89.827721   

       Precipitation Fister  Flow Kalltveit  Air Temperature Fister  
count          63288.000000    63288.000000            63288.000000  
mean               0.215365        6.509245                8.811811  
std                0.746016        7.565526                5.909750  
min                0.000000        0.000000              -11.800000  
25%                0.000000        2.683190                4.600000  
50%                0.000000        4.097010                8.400000  
75%                0.000000        7.141195               13.200000  
max               21.598444      110.256270               32.400000

Notes from the data:
- The maximum value for Q_Lyngsaana is much higher than the 75th percentile value, which may indicate the presence of outliers or extreme events in the data.
- The maximum value for Q_Kalltveit_uten_tapping is also much higher than the 75th percentile value.
- The minimum value for Air Temperature Fister is negative (-11.8), which may indicate an error in the data or a problem with the measurement equipment
- The maximum value for Precipitation Fister is much higher than the 75th percentile value, which may indicate the presence of outliers or extreme events in the data.

In [145]:
var_with_almost_max_count = ["Flow Tapping", "Flow Lyngsvatn Overflow"]

In [146]:
data[data[var_with_almost_max_count].isna().any(axis=1)]

Datetime  Wind Speed Nilsebu  Air Temperature Nilsebu  \
35973 2019-02-07 21:00:00                 7.6                     -1.5   
35974 2019-02-07 22:00:00                 6.2                     -1.3   

       Wind Direction Nilsebu  Relative Humidity Nilsebu  \
35973                    93.4                       99.1   
35974                    56.4                       99.6   

       Water Level Lyngsåna  Water Temperature Hiafossen  \
35973                 0.499                          0.4   
35974                 0.503                          0.4   

       Water Level Hiafossen  Air Temperature Fister  Precipitation Fister  \
35973                  0.963                     4.8                   0.0   
35974                  0.964                     4.4                   0.0   

       ...  SNOW_MELT_HBV  SNOW_SWE_HBV  Evap_HBV  SOIL_WAT_HBV  GR_WAT_HBV  \
35973  ...       0.003359    174.627315       0.0     82.774333    4.735968   
35974  ...       0.019787    174.605838       0.0     82.774925    4.725312   

       Q_Kalltveit_uten_tapping  Q_HBV_mean  Q_Lyngsaana  Vanntemp. Lyngsåna  \
35973                  0.366822    0.469942     0.250995                 0.6   
35974                  0.372462    0.466138     0.259335                 0.6   

       Vanntemp. Kalltveit elv  
35973                      1.4  
35974                      1.4  

[2 rows x 37 columns]

In [147]:
data[var_with_almost_max_count].iloc[35970:35980]

Flow Tapping  Flow Lyngsvatn Overflow
35970      1.592862                      0.0
35971      1.591902                      0.0
35972      1.591902                      0.0
35973           NaN                      NaN
35974           NaN                      NaN
35975      1.588988                      0.0
35976      1.588988                      0.0
35977      1.587701                      0.0
35978      1.587780                      0.0
35979      1.586323                      0.0

- Values can be turned to cero.

In [148]:
var_that_varying = ["Wind Speed Nilsebu", "Wind Direction Nilsebu", "Relative Humidity Nilsebu", "Air Temperature Nilsebu", "Water Temperature Kalltveit Kum", "Water Level Kalltveit", "Vanntemp. Lyngsåna"]

In [149]:
data[var_that_varying].describe()

Wind Speed Nilsebu  Wind Direction Nilsebu  Relative Humidity Nilsebu  \
count        62955.000000            63079.000000               63079.000000   
mean           -11.359088              213.033360                  64.444010   
std            373.072147              113.673486                  43.178686   
min          -7999.000000              -99.900000                 -99.900000   
25%              5.200000              133.200000                  51.500000   
50%              6.100000              215.600000                  82.600000   
75%              7.500000              325.900000                  96.700000   
max             22.100000              360.000000                  99.700000   

       Air Temperature Nilsebu  Water Temperature Kalltveit Kum  \
count             63102.000000                     63269.000000   
mean                  3.636294                       -25.726374   
std                   6.919231                       563.037133   
min                 -20.700000                     -9999.000000   
25%                  -0.800000                         4.500000   
50%                   3.100000                         5.700000   
75%                   8.600000                         7.762500   
max                  27.200000                        15.843750   

       Water Level Kalltveit  Vanntemp. Lyngsåna  
count           63274.000000        63283.000000  
mean              -26.232780            6.402902  
std               517.624893            5.343528  
min             -9999.000000           -0.773633  
25%                 0.526380            1.415820  
50%                 0.579130            5.084766  
75%                 0.673130           11.600000  
max                 2.275520           19.400000

In [150]:
data[data[var_that_varying].isna().any(axis=1)][35:]

Datetime  Wind Speed Nilsebu  Air Temperature Nilsebu  \
1667  2015-03-11 11:00:00                 NaN                      NaN   
1716  2015-03-13 12:00:00            2.200000                      4.4   
6986  2015-10-19 02:00:00                 NaN                      NaN   
7352  2015-11-03 08:00:00                 NaN                      NaN   
7354  2015-11-03 10:00:00                 NaN                      NaN   
...                   ...                 ...                      ...   
59915 2021-11-01 11:00:00                 NaN                      NaN   
59916 2021-11-01 12:00:00                 NaN                      NaN   
60441 2021-11-23 09:00:00            6.133333                      NaN   
62833 2022-03-03 01:00:00            4.700000                     -3.3   
62834 2022-03-03 02:00:00            5.700000                     -3.3   

       Wind Direction Nilsebu  Relative Humidity Nilsebu  \
1667                      NaN                        NaN   
1716                    168.4                       42.3   
6986                      NaN                        NaN   
7352                      NaN                        NaN   
7354                      NaN                        NaN   
...                       ...                        ...   
59915                     NaN                        NaN   
59916                     NaN                        NaN   
60441                     NaN                        NaN   
62833                    14.0                       69.6   
62834                   328.1                       66.1   

       Water Level Lyngsåna  Water Temperature Hiafossen  \
1667                  0.949                          1.1   
1716                    NaN                          1.4   
6986                  0.412                          6.0   
7352                  0.561                          6.2   
7354                  0.563                          6.2   
...                     ...                          ...   
59915                   NaN                          NaN   
59916                   NaN                          NaN   
60441                   NaN                          NaN   
62833                   NaN                          NaN   
62834                   NaN                          NaN   

       Water Level Hiafossen  Air Temperature Fister  Precipitation Fister  \
1667                   1.169                     6.5              0.000000   
1716                   1.018                     9.9              0.000000   
6986                   0.980                     4.0              0.000000   
7352                   0.950                     9.3              0.000000   
7354                   0.953                     9.3              0.000000   
...                      ...                     ...                   ...   
59915                    NaN                    12.6              0.000028   
59916                    NaN                    12.3              0.099972   
60441                    NaN                     6.8              0.200083   
62833                    NaN                     1.7              0.000000   
62834                    NaN                     3.3              0.000000   

       ...  SNOW_MELT_HBV  SNOW_SWE_HBV  Evap_HBV  SOIL_WAT_HBV  GR_WAT_HBV  \
1667   ...            NaN           NaN       NaN           NaN         NaN   
1716   ...            NaN           NaN       NaN           NaN         NaN   
6986   ...            NaN           NaN       NaN           NaN         NaN   
7352   ...            NaN           NaN       NaN           NaN         NaN   
7354   ...            NaN           NaN       NaN           NaN         NaN   
...    ...            ...           ...       ...           ...         ...   
59915  ...       0.000000      0.000000  0.027669     77.845670   26.679156   
59916  ...       0.000000      0.000000  0.024946     77.820895   26.294862   
60441  ...       0.077125      5.18475

In [151]:
data[var_that_varying].iloc[7330:7360]

Wind Speed Nilsebu  Wind Direction Nilsebu  Relative Humidity Nilsebu  \
7330                 5.4                   348.6                       88.3   
7331                 5.6                   352.7                       83.7   
7332                 5.6                   344.3                       78.9   
7333                 5.2                   336.3                       76.3   
7334                 5.3                   343.4                       71.2   
7335                 5.2                   351.5                       69.1   
7336                 6.5                   332.4                       80.6   
7337                 5.9                   338.6                       79.3   
7338                 5.7                   331.0                       81.9   
7339                 5.8                   347.2                       87.5   
7340                 6.4                   342.6                       88.2   
7341                 6.7                   338.6                       88.1   
7342                 5.9                   346.8                       86.8   
7343                 5.8                   335.7                       83.2   
7344                 6.2                   338.8                       82.8   
7345                 2.1                   331.9                       79.1   
7346                 1.9                   342.7                       83.4   
7347                 1.7                   336.5                       81.8   
7348                 6.0                   345.5                       83.2   
7349                 5.7                   328.9                       89.5   
7350                 6.0                   336.8                       91.8   
7351                 5.7                   353.9                       94.1   
7352                 NaN                     NaN                        NaN   
7353             -7999.0                   343.7                       83.0   
7354                 NaN                     NaN                        NaN   
7355                 NaN                     NaN                        NaN   
7356               -99.9                   -99.9                       79.1   
7357             -7999.0                   343.7                       82.9   
7358                 2.7                   176.0                       83.8   
7359                 1.4                   177.0                       86.3   

      Air Temperature Nilsebu  Water Temperature Kalltveit Kum  \
7330                      4.8                            7.310   
7331                      5.6                            7.310   
7332                      6.9                            7.310   
7333                      7.4                            7.310   
7334                      7.7                            7.310   
7335                      6.8                            7.310   
7336                      5.2                            7.310   
7337                      4.7                            7.310   
7338                      4.4                            7.310   
7339                      3.4                            7.310   
7340                      3.2                            7.310   
7341                      3.1                            7.310   
7342                      3.7                            7.265   
7343                      4.2                            7.250   
7344                      4.5                            7.250   
7345                      4.6                            7.250   
7346                      4.1                            7.250   
7347                      4.5                            7.250   
7348                      4.7                            7.250   
7349                      4.1                            7.250   
7350                      4.5                            7.250   
7351                      4.1                            7.250   
7352                      NaN         

- Wind Speed Nilsebu seem broken. Not to be included. -7999.0 and -99.9 NaN?
- Same with Wind Direction Nilsebu

In [152]:
data[["Relative Humidity Nilsebu", "Air Temperature Nilsebu", "Water Temperature Kalltveit Kum", "Water Level Kalltveit", "Vanntemp. Lyngsåna"]].describe()

Relative Humidity Nilsebu  Air Temperature Nilsebu  \
count               63079.000000             63102.000000   
mean                   64.444010                 3.636294   
std                    43.178686                 6.919231   
min                   -99.900000               -20.700000   
25%                    51.500000                -0.800000   
50%                    82.600000                 3.100000   
75%                    96.700000                 8.600000   
max                    99.700000                27.200000   

       Water Temperature Kalltveit Kum  Water Level Kalltveit  \
count                     63269.000000           63274.000000   
mean                        -25.726374             -26.232780   
std                         563.037133             517.624893   
min                       -9999.000000           -9999.000000   
25%                           4.500000               0.526380   
50%                           5.700000               0.579130   
75%                           7.762500               0.673130   
max                          15.843750               2.275520   

       Vanntemp. Lyngsåna  
count        63283.000000  
mean             6.402902  
std              5.343528  
min             -0.773633  
25%              1.415820  
50%              5.084766  
75%             11.600000  
max             19.400000

In [153]:
#TODO: Continue here with checking out the variables

In [154]:
var_with_HBV = ["GR_WAT_HBV", "SOIL_WAT_HBV", "Evap_HBV", "SNOW_SWE_HBV", "SNOW_MELT_HBV", "Q_HBV_mean"]

In [155]:
data[var_with_HBV]

GR_WAT_HBV  SOIL_WAT_HBV  Evap_HBV  SNOW_SWE_HBV  SNOW_MELT_HBV  \
0             NaN           NaN       NaN           NaN            NaN   
1             NaN           NaN       NaN           NaN            NaN   
2             NaN           NaN       NaN           NaN            NaN   
3             NaN           NaN       NaN           NaN            NaN   
4             NaN           NaN       NaN           NaN            NaN   
...           ...           ...       ...           ...            ...   
63283   12.691728     84.613501  0.000000    452.399884       0.016940   
63284   12.600009     84.613558  0.000171    452.361058       0.035770   
63285   12.541249     84.612921  0.001072    452.287223       0.068025   
63286   12.489559     84.612209  0.001187    452.206589       0.074288   
63287   12.424557     84.611742  0.000850    452.141680       0.059800   

       Q_HBV_mean  
0             NaN  
1             NaN  
2             NaN  
3             NaN  
4             NaN  
...           ...  
63283    2.401503  
63284    2.366949  
63285    2.341590  
63286    2.324290  
63287    2.308757  

[63288 rows x 6 columns]

In [159]:
var_with_water = ["Water Level Hiavatn", "Water Temperature Hiavatn", "Water Level Viglesdalsvatn", "Water Temperature Viglesdalsvatn", "Water Level Musdalsvatn Downstream", "Water Temperature Musdalsvatn Downstream", "Water Temperature Musdalsvatn", "Water Level Musdalsvatn", "Water Temperature Hiafossen", "Water Level Hiafossen", "Vanntemp. Kalltveit elv", "Water Level Lyngsåna"]

In [160]:
# remove rows with NaN values
data_1 = data[var_with_max_count+var_with_almost_max_count].dropna()

In [161]:
# remove rows with NaN values
data_2 = data[var_that_varying].dropna()

In [162]:
# remove rows with NaN values
data_3 = data[var_with_water].dropna()

In [163]:
# remove rows with NaN values
data_4 = data[var_with_HBV].dropna()

In [165]:
get_correlation_matrix(data_1)

Q_Lyngsaana  Q_Kalltveit_uten_tapping  TEMP_HBV  \
Q_Lyngsaana                  0.000000                  0.746794 -0.596699   
Q_Kalltveit_uten_tapping     0.746794                  0.000000 -0.401349   
TEMP_HBV                    -0.596699                 -0.401349  0.000000   
Precipitation Nilsebu        0.194985                 -0.451358  0.143385   
PRECIP_HBV                   0.200560                 -0.499174  0.154469   
Q_HBV                        0.639462                  0.914092 -0.344257   
Precipitation Fister         0.166631                 -0.558370  0.189603   
Flow Kalltveit               0.714289                  0.983200 -0.380909   
Air Temperature Fister      -0.653162                 -0.462841  0.954937   
Flow Tapping                 0.039608                 -0.625640  0.236367   
Flow Lyngsvatn Overflow      0.176624                 -0.561921  0.176762   

                          Precipitation Nilsebu  PRECIP_HBV     Q_HBV  \
Q_Lyngsaana                            0.194985    0.200560  0.639462   
Q_Kalltveit_uten_tapping              -0.451358   -0.499174  0.914092   
TEMP_HBV                               0.143385    0.154469 -0.344257   
Precipitation Nilsebu                  0.000000    0.949912 -0.467459   
PRECIP_HBV                             0.949912    0.000000 -0.510363   
Q_HBV                                 -0.467459   -0.510363  0.000000   
Precipitation Fister                   0.914087    0.956965 -0.546242   
Flow Kalltveit                        -0.485965   -0.535830  0.908459   
Air Temperature Fister                 0.171934    0.184572 -0.398421   
Flow Tapping                           0.841160    0.883868 -0.566647   
Flow Lyngsvatn Overflow                0.892212    0.938434 -0.513619   

                          Precipitation Fister  Flow Kalltveit  \
Q_Lyngsaana                           0.166631        0.714289   
Q_Kalltveit_uten_tapping             -0.558370        0.983200   
TEMP_HBV                              0.189603       -0.380909   
Precipitation Nilsebu                 0.914087       -0.485965   
PRECIP_HBV                            0.956965       -0.535830   
Q_HBV                                -0.546242        0.908459   
Precipitation Fister                  0.000000       -0.594307   
Flow Kalltveit                       -0.594307        0.000000   
Air Temperature Fister                0.221884       -0.445676   
Flow Tapping                          0.890024       -0.567405   
Flow Lyngsvatn Overflow               0.945068       -0.586237   

                          Air Temperature Fister  Flow Tapping  \
Q_Lyngsaana                            -0.653162      0.039608   
Q_Kalltveit_uten_tapping               -0.462841     -0.625640   
TEMP_HBV                                0.954937      0.236367   
Precipitation Nilsebu                   0.171934      0.841160   
PRECIP_HBV                              0.184572      0.883868   
Q_HBV                                  -0.398421     -0.566647   
Precipitation Fister                    0.221884      0.890024   
Flow Kalltveit                         -0.445676     -0.567405   
Air Temperature Fister                  0.000000      0.273760   
Flow Tapping                            0.273760      0.000000   
Flow Lyngsvatn Overflow                 0.210796      0.898483   

                          Flow Lyngsvatn Overflow  
Q_Lyngsaana                              0.176624  
Q_Kalltveit_uten_tapping                -0.561921  
TEMP_HBV                                 0.176762  
Precipitation Nilsebu                    0.892212  
PRECIP_HBV                               0.938434  
Q_HBV                                   -0.513619  
Precipitation Fister                     0.945068  
Flow Kalltveit                          -0.586237  
Air Temperature Fister                   0.210796  
Flow Tapping                             0.898483  
Flow Lyngsvatn Overflow                  0.000000